warehouse/query-core/src/main/java/datawave/query/util/sortedset/MultiSetBackedSortedSet.java

```diff
@@ -141,26 +145,38 @@ public SortedSet<E> tailSet(E fromElement) {
     }
 
     @Override
-    public E first() {
+    public E first() throws NoSuchElementException {
+        if (sets == null || !sets.iterator().hasNext()) {
+            throw new NoSuchElementException("No elements in input sets");
+        }
         SortedSet<E> firstSet = new TreeSet<>(comparator());
         for (SortedSet<E> set : sets) {
-            E s = set.first();
-            if (s != null) {
+            if (set != null && !set.isEmpty()) {
+                E s = set.first();
                 firstSet.add(s);
             }
         }
+        if (firstSet.isEmpty()) {
+            throw new NoSuchElementException("No elements in input sets");
+        }
         return firstSet.first();
     }
 
     @Override
-    public E last() {
+    public E last() throws NoSuchElementException {
+        if (sets == null || !sets.iterator().hasNext()) {
```

Comment:
sets.iterator().hasNext() vs sets.isEmpty().  Why not go with the simpler call?

Reply:
Good catch! Updating